aten/src/ATen/native/native_functions.yaml

```diff
@@ -637,15 +637,41 @@
   dispatch:
     CPU: bmm_cpu
     CUDA: bmm_cuda
+    SparseCPU: bmm_sparse_cpu
+    SparseCUDA: bmm_sparse_cuda
+  supports_named_tensor: True
+
+- func: bmm.deterministic(Tensor self, Tensor mat2, bool deterministic) -> Tensor
```

Comment:
How about let's make the deterministic argument kwarg only. So `Tensor mat2, *, bool deterministic=False`?

Reply:
That means I'd have to have a TensorOptions argument in my C++ implementation right? Do you think I should remove the `bmm.deterministic` version and just add TensorOptions to every bmm function?

Reply:
Or maybe I could make it an c10::optional argument, like `Tensor mat2, *, bool? deterministic=False`.

Reply:
When I change this line to `- func: bmm.deterministic(Tensor self, Tensor mat2, *, bool deterministic=False) -> Tensor` and rebuild, I get this error:

```
Writing torch/csrc/autograd/generated/python_variable_methods.cpp
Traceback (most recent call last):
  File "tools/setup_helpers/generate_code.py", line 93, in <module>
    main()
  File "tools/setup_helpers/generate_code.py", line 88, in main
    options.selected_op_list_path,
  File "tools/setup_helpers/generate_code.py", line 43, in generate_code
    gen_autograd_python(declarations_path or DECLARATIONS_PATH, autograd_gen_dir, 'tools/autograd')
  File "/home/kurtamohler/development/pytorch-bmm/tools/autograd/gen_autograd.py", line 234, in gen_autograd_python
    out, aten_decls + deprecated, template_path)
  File "/home/kurtamohler/development/pytorch-bmm/tools/autograd/gen_python_functions.py", line 173, in gen_py_torch_functions
    env = create_python_bindings(py_torch_functions, is_python_method=False, is_module=False)
  File "/home/kurtamohler/development/pytorch-bmm/tools/autograd/gen_python_functions.py", line 192, in create_python_bindings
    py_methods.append(method_impl(name, overload_decls, is_python_method, is_module))
  File "/home/kurtamohler/development/pytorch-bmm/tools/autograd/gen_python_functions.py", line 884, in method_impl
    grouped = group_overloads(declarations, is_python_method)
  File "/home/kurtamohler/development/pytorch-bmm/tools/autograd/gen_python_functions.py", line 1046, in group_overloads
    x, list(dictionary.keys())))
RuntimeError: 'base' not in dictionary for bmm(Tensor input, Tensor mat2, bool deterministic). keys are ['out', 'signature']
```

I've been trying different things to get it working, but can't figure out it.

Reply:
OK, looks like the codegen doesn't actually understand how to deal with this :/ Optional positional arg it is, I suppose. I'll file an issue about this.

Reply:
Filed at https://github.com/pytorch/pytorch/issues/33547

Reply:
Cool thanks!